server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -662,14 +656,20 @@ public void close() throws IOException {
     }
 
     /**
-     * @return Whether a field is a metadata field.
+     * @return Whether a field is a metadata field for older indexes
+     * This is an outdated method and should not be used.
+     * TODO: remove in v 9.0, as all DocumentField objects will contain information whether they are metaField or not
      */
-    public static boolean isMetadataField(String fieldName) {
-        return META_FIELDS.contains(fieldName);
+    public static boolean isMetadataFieldSt(String fieldName) {
+        return IndicesModule.getBuiltInMetadataFields().contains(fieldName);
```

Comment:
META_FIELDS contained "_size" from the Mapper-Size plugin as a meta data field, which is not part of the built in fields. I don't know how much this matters but wanted to ask if this might need to be added for bwc?

Reply:
Also I assume "_ttl" and "_timestamp" are okay to remove as the comment above says, so I'm not digging further there...